packages/edit-post/src/components/sidebar/post-template/form.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+import { __experimentalInspectorPopoverHeader as InspectorPopoverHeader } from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
+import { addCard } from '@wordpress/icons';
+import { Notice, SelectControl, Button } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import PostTemplateCreateModal from './create-modal';
+
+export default function PostTemplateForm( {
+	isPostsPage,
+	selectedOption,
+	options,
+	canCreate,
+	canEdit,
+	onChange,
+	onEdit,
+	onClose,
+} ) {
+	const [ isCreateModalOpen, setIsCreateModalOpen ] = useState( false );
+
+	return (
+		<>
+			<InspectorPopoverHeader
+				title={ __( 'Template' ) }
+				help={ __(
+					'Templates define the way your content is displayed when viewing your site.'
+				) }
+				actions={
+					canCreate
+						? [
+								{
+									icon: addCard,
+									label: __( 'Add template' ),
+									onClick: () => setIsCreateModalOpen( true ),
+								},
+						  ]
+						: []
+				}
+				onClose={ onClose }
+			/>
+			{ isPostsPage ? (
+				<Notice
+					className="edit-post-post-template__notice"
+					status="warning"
+					isDismissible={ false }
+				>
+					{ __( 'The posts page template cannot be changed.' ) }
+				</Notice>
```

Comment:
This doesn't work because of the missing backport (https://github.com/WordPress/gutenberg/pull/38607#issuecomment-1138526934), but you can use the following snippet for testing.

```php
add_action( 'init', function() {
	register_setting(
		'reading',
		'page_for_posts',
		array(
			'show_in_rest' => true,
			'type'         => 'number',
			'description'  => __( 'The ID of the page that should display the latest posts', 'gutenberg' ),
		)
	);
} );
```

Reply:
Thanks!